hack/update-golangci-lint.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Copyright 2021 The Cockroach Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+if [[ -n "${BUILD_WORKSPACE_DIRECTORY:-}" ]]; then # Running inside bazel
+  echo "Linting go source files..." >&2
+elif ! command -v bazel &>/dev/null; then
+  echo "Install bazel at https://bazel.build" >&2
+  exit 1
+else
+  (
+    set -o xtrace
+    bazel run //hack:update-golangci-lint
+  )
+  exit 0
+fi
+
+golangci_lint=$(realpath "$1")
+
+cd "$BUILD_WORKSPACE_DIRECTORY"
+
+export GO111MODULE=on
```

Comment:
We're using go 1.16 so I don't think this is necessary

Reply:
Removed export statement from this file. 
However, that statement is currently used in multiple files. Might need to remove from those files as well.